src/timesync/timesyncd-manager.c

```diff
@@ -121,7 +121,7 @@ static int manager_send_request(Manager *m) {
          */
         assert_se(clock_gettime(clock_boottime_or_monotonic(), &m->trans_time_mon) >= 0);
         assert_se(clock_gettime(CLOCK_REALTIME, &m->trans_time) >= 0);
-        ntpmsg.trans_time.sec = htobe32(m->trans_time.tv_sec + OFFSET_1900_1970);
+        ntpmsg.trans_time.sec = htobe32(time_to_uint32(m->trans_time.tv_sec + OFFSET_1900_1970));
```

Comment:
so the addition is likely going to overflow, on systems where time_t is 32bit. hence maybe:

```c
static uint32_t graceful_add_offset_1900_1970(time_t t) {
        /* Adds OFFSET_1900_1970 to t and returns it as 32bit value. This is handles overflows
         * gracefully in a deterministic and well-defined way by cutting off the top bits. */
        uint64_t a = (uint64_t) t + OFFSET_1900_1970;
        return (uint32_t) (a & UINT64_C(0xFFFFFFFF));
}
```